### DISCUSSION TITLE
Allow importing and exporting of settings

### DISCUSSION THREAD

**Discussion by elydpg** [breezy-weather/breezy-weather#1100] [Ideas]
### Describe your idea

Pretty much the title. Breezy weather has a lot of different options both for the app as a whole and for each individual location. Thus I propose that the settings allow an option to export them (as say a JSON file) and import them, which would be helpful for testing or for setting the app up on a new phone

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Duplicate of #231. Locations can be exported but there will be no option to import, as we cannot guarantee the integrity and consistency of what we import.

**Reply by elydpg** [reply]
Ah dang, I forgot to check the ideas and I only checked issues. I don't currently see a way that locations can be exported at all, or are you referring to what is planned to be introduced with #231 ?

**Reply by papjul** [reply]
No, it's not planned, what's your use case? Would #863 fits your needs?

**Reply by elydpg** [reply]
I mean it seems like a low priority feature in any case. For me it's more I like to keep track of all the settings I've changed, including the list of locations and all of their settings as well, but even without the location list and their settings it's still a lot to keep track of, especially since the app remembers widget settings even when I delete the widget

**Reply by papjul** [reply]
Can't you take a screenshot of your location list? It seems more convenient than a json file.

Export of widgets is not planned because widgets don't persist after an app is uninstalled, so we can't restore them, and the overhaul of widgets #937 will make the preferences tied to each added widget to allow to use the same widget with different locations

**Reply by elydpg** [reply]
I suppose, but that doesn't help for longer lists nor for highly customized sourcing, modelling, and theming settings. 

I assume that any issues pertaining to the widgets will become moot after the widget overhaul, so I'll test those more carefully once that's been done. Currently I'm using the confusingly named "daily" widget to replicate the current weather widget in the stock Samsung weather app and it's working fine 😌
